util/include/crypto/openssl/digest_creator.hpp

```diff
@@ -12,13 +12,14 @@
 #pragma once
 
 #include "hash.hpp"
+#include "crypto/crypto.hpp"
 #include "crypto/digest_creator.hpp"
 
 namespace concord::crypto::openssl {
 
 template <typename SHACTX,
-          typename = std::enable_if_t<std::is_same_v<SHACTX, concord::crypto::openssl::SHA2_256> ||
-                                      std::is_same_v<SHACTX, concord::crypto::openssl::SHA3_256>>>
+          typename = std::enable_if_t<std::is_same_v<SHACTX, concord::crypto::SHA2_256> ||
```

Comment:
You should use the internal name here (openssl::EVPHash<EVP_sha3_256, 32>) because this code is in the openssl directory.

Reply:
From my point of view this is correct since we wanted to use hashing algorithm wrappers: `SHA2_256` & `SHA3_256`.

The alternate way would be to write as below:
`typename = std::enable_if_t<std::is_same_v<SHACTX, openssl::EVPHash<EVP_sha256, 32>>`

Reply:
Your code is correct, because it passes the tests, but this file is the internal openssl implementation of the digestcreator interface, and thus should use openssl types directly and not aliases.
If we decide to add the option of using another library for hashing, this code will no longer compile (assuming that we also change the default concord::crypto::SHA2_256).